presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/thrift/ThriftHiveMetastore.java

```diff
@@ -503,16 +509,16 @@ private void deleteTableColumnStatistics(HiveIdentity identity, String databaseN
     @Override
     public synchronized void updatePartitionStatistics(HiveIdentity identity, String databaseName, String tableName, String partitionName, Function<PartitionStatistics, PartitionStatistics> update)
     {
-        PartitionStatistics currentStatistics = requireNonNull(
-                getPartitionStatistics(identity, databaseName, tableName, ImmutableSet.of(partitionName)).get(partitionName), "getPartitionStatistics() returned null");
-        PartitionStatistics updatedStatistics = update.apply(currentStatistics);
-
         List<Partition> partitions = getPartitionsByNames(identity, databaseName, tableName, ImmutableList.of(partitionName));
         if (partitions.size() != 1) {
             throw new PrestoException(HIVE_METASTORE_ERROR, "Metastore returned multiple partitions for name: " + partitionName);
         }
-
         Partition originalPartition = getOnlyElement(partitions);
+
+        PartitionStatistics currentStatistics = requireNonNull(
+                getPartitionStatistics(identity, databaseName, tableName, ImmutableList.of(originalPartition)).get(partitionName), "getPartitionStatistics() returned null");
```

Comment:
s/`ImmutableList.of(originalPartition)`/`partitions`

Reply:
Also, the message `getPartitionStatistics() returned null` is wrong. 
It should rather say "getPartitionStatistics() did not return statistics for partition".
And it should better be `checkState`. 

(I know you only moved this code, but worth fixing anyway)